atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/PostgresKeyValueServiceConfig.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2015 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Supplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.DbTableFactory;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.PostgresDbTableFactory;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.SqlConnectionSupplier;
+
+@JsonDeserialize(as = ImmutablePostgresKeyValueServiceConfig.class)
+@JsonSerialize(as = ImmutablePostgresKeyValueServiceConfig.class)
+@JsonTypeName(PostgresKeyValueServiceConfig.TYPE)
+@Value.Immutable
+public abstract class PostgresKeyValueServiceConfig extends DbKeyValueServiceConfig {
+
+    public static final String TYPE = "postgres";
+
+    @Override
+    public Supplier<DbTableFactory> tableFactorySupplier() {
+        return () -> new PostgresDbTableFactory(shared());
+    }
+
+    @Value.Default
+    @Override
+    public SqlConnectionSupplier sqlConnectionSupplier() {
+        throw new RuntimeException("unimplemented -- waiting on pooling");
```

Comment:
I have a commit on top of this that wipes all of this out anyways so I'll merge this through without edit.